src/test/java/io/neonbee/test/base/NeonBeeTestBase.java

```diff
@@ -113,26 +143,42 @@ public void setUp(Vertx vertx, VertxTestContext testContext, TestInfo testInfo)
             } else {
                 neonBee = asyncNeonBee.result();
 
-                DeploymentOptions serverVerticleOptions = readDeploymentOptions(ServerVerticle.class, workingDirPath);
+                if (customUserPrincipal.get()) {
+                    // use a dummy ServerVerticle that returns the principal by calling the provideUserPrincipal method
+                    DeploymentOptions serverVerticleOptions =
+                            readDeploymentOptions(ServerVerticle.class, workingDirPath);
 
-                Optional.ofNullable(provideUserPrincipal(testInfo)).map(userPrincipal -> {
-                    // Replace current ServerVerticle with a dummy ServerVerticle that also has a dummy AuthHandler to
-                    // provide the user principal specified in the provideUserPrincipal method
-                    ServerVerticle dummyServerVerticle = createDummyServerVerticle(testInfo);
+                    try {
+                        // just to be sure, try to fetch a new port to use
+                        options.setServerPort(SystemHelper.getFreePort());
```

Comment:
The downside of this is, that the ServerVerticleConfig  in /config folder is not longer in sync with the instantiated object .. Maybe we should write the "new" config also back into the /config foler.

Reply:
Yes I knew. I tolerated this, because this is only covering an edge case, which should normally not ocour. However all methods which read the port, anyways read it from the options. The config is just there "in addition". So we should be good with this.